### PR TITLE
fix: restrict age field's input

### PR DESCRIPTION
**Pull request by jamil314**
## Description

Clearly describe what has been changed. Include relevant context or background.
Explain how the issue was fixed (if applicable) and the root cause.

Link this pull request to the GitHub issue (and optionally name the branch `ocrvs-<issue #>`)

[e2e PR](https://github.com/opencrvs/opencrvs-farajaland/pull/1787)

## Checklist

- [ ] I have linked the correct Github issue under "Development"
- [ ] I have tested the changes locally, and written appropriate tests
- [ ] I have tested beyond the happy path (e.g. edge cases, failure paths)
- [ ] I have updated the changelog with this change (if applicable)
- [ ] I have updated the GitHub issue status accordingly

### CODE REVIEW

File: packages/client/src/v2-events/features/events/registered-fields/AgeField.tsx [hunk 1]
```diff
@@ -27,15 +27,37 @@ function AgeInput({ asOfDateRef, ...props }: AgeInputProps) {
...
+        }
+      }}
+      onPaste={(e: React.ClipboardEvent<HTMLInputElement>) => {
+        const text = e.clipboardData.getData('text')
+        const numeric = Number(text)
+        e.currentTarget.value = String(numeric)
+        e.preventDefault()
+        props.onChange(numeric ? { age: numeric, asOfDate } : undefined)
+      }}
     />
```
Comment (jamil314) [inline packages/client/src/v2-events/features/events/registered-fields/AgeField.tsx:54]: Number field's inputs translates to number before storing it in the form.
`'00032'` becomes `32`
`'1e2'` becomes `100`
So, field validation does not catch these.

File: packages/client/src/v2-events/features/events/registered-fields/AgeField.tsx [hunk 2]
```diff
@@ -27,15 +27,37 @@ function AgeInput({ asOfDateRef, ...props }: AgeInputProps) {
...
+      }}
       onChange={(newAge) =>
         props.onChange(
           newAge === undefined ? undefined : { age: newAge, asOfDate }
         )
       }
+      onKeyDownCapture={(e: React.KeyboardEvent<HTMLInputElement>) => {
+        if (
+          ['.', 'e'].includes(e.key) ||
+          (e.key === '0' && !e.currentTarget.value)
```
Comment (makelicious) [inline packages/client/src/v2-events/features/events/registered-fields/AgeField.tsx]: Could we do it positively? I suppose this is close to what we want? `[1-9]{1,3}`

File: packages/client/src/v2-events/features/events/registered-fields/AgeField.tsx (hunk 1 above)
Comment (jamil314) [reply packages/client/src/v2-events/features/events/registered-fields/AgeField.tsx:54]: @Zangetsu101  We can not show errors because validation does not pick them up.

File: packages/client/src/v2-events/features/events/registered-fields/AgeField.interaction.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+        }}
+      />
+    )
+  },
+  play: async ({ canvasElement }) => {
+    const canvas = within(canvasElement)
+
+    const ageInput = await canvas.findByTestId('number__storybook____age')
+    await userEvent.clear(ageInput)
+    await userEvent.type(ageInput, '0020')
```
Comment (makelicious) [inline packages/client/src/v2-events/features/events/registered-fields/AgeField.interaction.stories.tsx]: I wrote some tests for phone number and noticed this

```
      // NOTE: Using userEvent does not work for number inputs. It'll remove the leading 0 for you.
      await fireEvent.change(input, { target: { value: '01741234567' } })
```

I tried changing this to fireEvent. Test passes but it seems to allow it

Image

File: packages/client/src/v2-events/routes/routes.stories.tsx [hunk 4]
```diff
@@ -34,7 +34,7 @@ export const Team: StoryObj<typeof UserList> = {
   parameters: {
     userRole: TestUserRole.enum.REGISTRATION_AGENT,
     reactRouter: {
-      initialPath: `/v2/team/users?locationId=${generator.user.registrationAgent().v2.primaryOfficeId}`,
+      initialPath: `/team/users?locationId=${generator.user.registrationAgent().v2.primaryOfficeId}`,
```
Comment (makelicious) [inline packages/client/src/v2-events/routes/routes.stories.tsx:37]: are these from some other pr?

File: packages/client/src/v2-events/features/events/registered-fields/AlphaPrintButton.tsx [hunk 5]
```diff
@@ -62,7 +62,7 @@ export const AlphaPrintButton = {
     const intl = useIntl()
     const location = useLocation()
     const parts = location.pathname.split('/')
-    const eventId = UUID.parse(parts[4])
+    const eventId = UUID.parse(parts[3])
```
Comment (makelicious) [inline packages/client/src/v2-events/features/events/registered-fields/AlphaPrintButton.tsx:65]: is this from other pr?
